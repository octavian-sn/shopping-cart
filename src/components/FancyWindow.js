import React from 'react';
import '../styles/fancy-window.css';
import PropTypes from 'prop-types';

function FancyWindow({ size, children }) {
  return (
    <div className={`fancy-window ${size}`}>
      <div className="upper border"></div>
      <div className="content">{children}</div>
      <div className="lower border"></div>
    </div>
  );
}

export default FancyWindow;

// FancyWindow.defaultProps = { size: 'big' };
FancyWindow.propTypes = { size: PropTypes.oneOf(['big', 'small']).isRequired };
