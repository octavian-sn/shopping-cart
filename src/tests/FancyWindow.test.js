import React from 'react';
import { render } from '@testing-library/react';
import FancyWindow from '../components/FancyWindow';

describe('FancyWindow component', () => {
  it('renders a continue button', () => {
    const { container } = render(<FancyWindow />);
    expect(container).toMatchSnapshot();
  });
});
