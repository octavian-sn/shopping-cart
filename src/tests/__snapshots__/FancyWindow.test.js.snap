// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyWindow component renders a continue button 1`] = `
<div>
  <div
    class="fancy-window undefined"
  >
    <div
      class="upper border"
    />
    <div
      class="content"
    />
    <div
      class="lower border"
    />
  </div>
</div>
`;
